core/src/main/java/org/infinispan/commands/write/PutMapCommand.java

```diff
@@ -251,8 +261,8 @@ public boolean ignoreCommandOnStatus(ComponentStatus status) {
    }
 
    @Override
-   public boolean readsExistingValues() {
-      return false;
+   public LoadType loadType() {
+      return hasFlag(Flag.IGNORE_RETURN_VALUES) ? LoadType.DONT_LOAD : LoadType.PRIMARY;
```

Comment:
Well, `QueryInterceptor` doesn't really care about the return value, it needs the previous value -- it's just that with the "classic" single-key cache operations, the return value was enough to determine the previous value. If a functional operation replaces "a" with "b" and returns "c", we must remove "a" from the index, add "b" instead, and ignore "c".

We sometimes need to force a remote invocation to return the previous value to the originator for indexing, but I'm hoping we can separate that from the command's return value, so we can implement it the same way for each command instead of having ever more elaborate conventions for the return values.